src/components/utils/src/socket_posix.cc

```diff
@@ -36,191 +36,262 @@
+  if (!IsValid()) {
+    return -1;
   }
-  return *this;
+  const int flags = MSG_NOSIGNAL;
+  int result = send(tcp_socket_, buffer, size, flags);
+  if (-1 == result) {
+    LOG4CXX_ERROR(logger_ptr, "Failed to send data: " << errno);
+    return -1;
+  }
+  return result;
 }
 
-bool utils::Socket::Valid() const {
-  return impl_->Valid();
+bool
+utils::TcpSocketConnection::Impl::Close() {
+  return CloseSocket(tcp_socket_);
 }
 
-bool utils::Socket::Create(int af, int type, int protocol) {
-  return impl_->Create(af, type, protocol);
+bool
+utils::TcpSocketConnection::Impl::IsValid() const {
+  return tcp_socket_ != NULL;
 }
 
-bool utils::Socket::Close() {
-  return impl_->Close();
-}
+////////////////////////////////////////////////////////////////////////////////
+/// utils::TcpSocketConnection
+////////////////////////////////////////////////////////////////////////////////
 
-bool utils::Socket::SetOptions(
-  int level, int optname, const char* optval, size_t optlen) {
-  return impl_->SetOptions(level, optname, optval, optlen);
+utils::TcpSocketConnection::TcpSocketConnection()
+  : impl_(new Impl()) {
 }
 
-bool utils::Socket::Bind(
-  const struct sockaddr* addr, size_t addrlen) {
-  return impl_->Bind(addr, addrlen);
+utils::TcpSocketConnection::TcpSocketConnection(Impl* impl)
+  : impl_(impl) {
+  DCHECK(impl_);
 }
 
-bool utils::Socket::Listen(int backlog) {
-  return impl_->Listen(backlog);
+utils::TcpSocketConnection::~TcpSocketConnection() {
+  delete impl_;
 }
 
-utils::Socket utils::Socket::Accept(
-  struct sockaddr* addr, size_t* addrlen) {
-  return impl_->Accept(addr, addrlen);
+utils::TcpSocketConnection::TcpSocketConnection(TcpSocketConnection& rhs) {
+  Swap(rhs);
 }
 
-ssize_t utils::Socket::Send(
-  const char* buf, size_t length, int flags) {
-  return impl_->Send(buf, length, flags);
+utils::TcpSocketConnection&
+utils::TcpSocketConnection::operator=(TcpSocketConnection& rhs) {
+  if (this != &rhs) {
+    Swap(rhs);
+  }
+  return *this;
 }
 
-utils::Socket::Socket(Socket::Impl* impl)
-  : impl_(impl) {
+void
+utils::TcpSocketConnection::Swap(TcpSocketConnection& rhs) {
+  using std::swap;
+  swap(impl_, rhs.impl_);
 }
 
-void utils::Socket::Swap(Socket& rh) {
-  std::swap(this->impl_, rh.impl_);
+ssize_t utils::TcpSocketConnection::Send(const char* buffer, std::size_t size) {
+  return impl_->Send(buffer, size);
 }
 
-utils::Socket::Impl::Impl(): socket_(NULL) {
+bool utils::TcpSocketConnection::Close() {
+  return impl_->Close();
 }
 
-utils::Socket::Impl::Impl(int socket)
-  : socket_(socket <= 0 ? NULL : socket) {
+bool utils::TcpSocketConnection::IsValid() const {
+  return impl_->IsValid();
 }
 
-utils::Socket::Impl::~Impl() {
-  Close();
-}
+////////////////////////////////////////////////////////////////////////////////
+/// utils::ServerTcpSocket::Impl
+////////////////////////////////////////////////////////////////////////////////
+
+class utils::TcpServerSocket::Impl {
+public:
```

Comment:
I'll add the space, but I don't to bother with the `ccplint/astyle` right now. We have a plan to use the clang-format (not the astyle) for all sources, I don't to waste time at the moment with  `ccplint/astyle` configuration.